src/Input/Item.hs

```diff
@@ -167,6 +181,11 @@ splitUsing f = repeatedly $ \(x:xs) ->
     let (a,b) = break (isJust . f) xs
     in ((fromMaybe mempty $ f x, x:a), b)
 
+item_test :: IO ()
+item_test = testing "Input.Item.Target JSON (encode . decode = id) " $ do
+  quickCheck $ \(t :: Target) -> case J.eitherDecode $ J.encode t of
```

Comment:
This could instead be written as:

`quickCheck $ \(t :: Target) -> J.eitherDecode (J.encode t) == Right t`